### DISCUSSION TITLE
Matter via OTA

### DISCUSSION THREAD

**Discussion by jpadie** [arendst/Tasmota#19094] [Q&A]
I've read that matter is part of the binary for all installs via the web-installer.  I've updated a dozen or so of my esp32 tasmota estate to 13.0.0 via OTA (they are in walls and other inaccessible places) but do not see the matter configuration menu item. 

Is there a (n approved) way to enable matter via OTA upgrades?  

thanks
Justin

**Comment by s-hadinger** [comment]
`SetOption151 1`

And you should see a Matter button 

<img width="375" alt="image" src="https://github.com/arendst/Tasmota/assets/49731213/f6d647de-7c58-472f-9bc0-78df818d10ed">

**Reply by jpadie** [reply]
Merci Stephan
but unfortunately the menu does not appear 


```
Program Version | 13.0.0(display)
-- | --
2023-06-26T10:55:05
2_0_10/4.4.5.230614
```


![Untitled](https://github.com/arendst/Tasmota/assets/25035905/121d090f-4a07-4e7a-bb7a-7398bc90e8d5)
![Untitled](https://github.com/arendst/Tasmota/assets/25035905/9ce9041a-cb7d-4470-b8f8-2ee600823cd0)

**Reply by s-hadinger** [reply]
I think it's included in the standard Tasmota32 but not in display variant. @Jason2866  ?

**Reply by jpadie** [reply]
ah.  I was looking at this text on the [website](https://tasmota.github.io/docs/Matter/#supported-platforms):

> Matter is supported by default. The binaries are available for all ESP32 variants in our [web installer](https://tasmota.github.io/install/)

hence my question about OTA vs webInstaller.   If it's not available in the display variant then I can try compiling a version and installing over the web.  I'd assumed (given the web-installer reference) that there might need to be tethering for the initial installation with matter.

**Reply by Jason2866** [reply]
Matter is not in Display variant. Display is already a big firmware, adding matter would make it hough.
So it is not included. 
You can flash your self built factory firmware with this enhanced WebInstaller https://jason2866.github.io/Tasmota-specials/

**Reply by jpadie** [reply]
thanks Jason - I don't think that the [page here ](https://tasmota.github.io/docs/Matter/#supported-platforms) is set up as a wiki thus not user-editable.  Do you think it worthwhile to change the text so that it is clearer that not all binaries contain matter?

on reflection, flashing my own builds is not the answer here.  I need most of the devices to stay on stock builds.  As many use displays I think that means I either need to build an mqtt to matter bridge or avoid using matter.  I'd hoped to be able to use it so as to avoid writing my own code for an Alexa implementation on the various parts of the building control system.

**Reply by sfromis** [reply]
That page is actually editable wiki-style, pencil top right.

What's meant by "all ESP32 variants" is as in both ESP32, ESP32-S2, ESP32-S3 and ESP32-C3 - not _Tasmota_ variants within each ESP32 variant.

In many cases, you can use the existing Tasmota bridging meant to provide support for ESP8266 devices (which cannot get Tasmota Matter), to also support ESP32 devices without Matter included in the build.

**Reply by Jason2866** [reply]
So you have a setup where a Display and a sensor is connected and you want to provide the sensor data via matter? So imho it is a good idea to do this with an extra matter bridge device as sfromis suggested. No need to self compile.

**Reply by jpadie** [reply]
thanks all.

a use case is for some devices which are in-wall and have a display. their purpose is to report the temperature and to be able to change the set point and hysteresis for heating/cooling.  the devices don't do any switching, just display interaction and reporting the temps.  Temps are currently shared via mqtt. 
I use an external mqtt server which is causing some problems due to fibre installers all over my village.  outages of several hours at a time are not unusual.
the use case for matter would be to report info locally (i.e. temps and changes in set point) and also to allow interaction with alexa to (i) change the set point as an alexa command and (ii) get the temp as an alexa response.  
It looks like this is possible with matter.  but as I've not played with it yet, I may well be wrong!

**Reply by sfromis** [reply]
When the aim includes cross-device automation, in addition to what you can do with Matter, I'd much recommend having a local MQTT broker. Can be a pretty cheap system, and is also a feature often seen on NAS boxes and maybe routers. That bad outages you have is not really suitable for home automation, without everything being local.

**Reply by jpadie** [reply]
@sformis: yes but <as ever> ...   I am mostly remote from my house.  having some ability to interact with the tech in the house is important.  my vpn end point seems temperamental so I tend to rely on setting retained instructions via mqtt.  I know I could set up an mqtt relay - it's on the list of things to do ...
